npm/src/language-service/language-service.ts

```diff
@@ -142,7 +147,7 @@ export class QSharpLanguageService implements ILanguageService {
   }
 
   async updateConfiguration(config: IWorkspaceConfiguration): Promise<void> {
```

Comment:
Why is this function async when we don't await?

Reply:
Same answer for all your comments: `ILanguageService`, which this class implements, has to define all of its methods as `async`. This is because `ILanguageService` can be proxied to a web worker (same as the debugger and compiler). So even though the implementation you're looking at is synchronous, the _interface_ needs to be async, to account for the fact that we may be running the language service in a worker.